lib/error_catalog.h

```diff
@@ -97,10 +103,19 @@ class ErrorCatalog {
 
     /// add to the catalog
     /// returns false if the code already exists and forceReplace was not set to true
+    /// @param type      - error/warning/info message
     /// @param errorCode - integer value for the error/warning
     /// @param name      - name for the error. Used to enable/disable all errors of that type
     /// @param forceReplace - override an existing error type in the catalog
-    bool add(int errorCode, const char *name, bool forceReplace = false) {
+    template <MessageType type, int errorCode>
+    bool add(const char *name, bool forceReplace = false) {
+        static_assert(type != MessageType::Error ||
```

Comment:
Neat, does this work `errorcode` or should we make explicit that it is a `constexpr`? Same for all the error codes.

Reply:
Sorry, what do you mean?

Reply:
All the integers are `const` not `constexpr`. Does this work with `static_assert`? Or does the compiler convert them automatically to compile time constants?

Reply:
Yes, it works. The compilation fails with:
```
lib/error_catalog.h:112:50: error: static assertion failed
  112 |         static_assert(type != MessageType::Error ||
      |                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
  113 |             (errorCode >= ErrorType::ERR_MIN_BACKEND && errorCode <= ErrorType::ERR_MAX));
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
I believe that for an integral type `const` implies `consexpr`.